### PR TITLE
Shaokai/release 0.1.2

### PR DESCRIPTION
**Pull request by yeshaokai**
Supports 3D pose
Better tests
More flexible identifier

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install pytest numpy==1.23.5 tables==3.8.0
+        pip install git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[modelzoo]
```
Comment (MMathisLab) [inline .github/workflows/pytest.yml:39]: best to use pip; rc4?

File: README.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ In our original work (NeurIPS 2023) we used GPT3.5 and GPT4 as part of our agent
 Conda is an easy-to-use Python interface that supports launching [Jupyter Notebooks](https://jupyter.org/). If you are completely new to this, we recommend checking out the [docs here for getting conda installed](https://deeplabcut.github.io/DeepLabCut/docs/beginner-guides/beginners-guide.html#beginner-user-guide). Otherwise, proceed to use one of [our supplied conda files](https://github.com/AdaptiveMotorControlLab/AmadeusGPT/tree/main/conda). As you will see we have minimal dependencies to get started, and [here is a simple step-by-step guide](https://deeplabcut.github.io/DeepLabCut/docs/installation.html#step-2-build-an-env-using-our-conda-file) you can reference for setting it up (or see [BONUS](README.md#bonus---customized-your-conda-env) below). Here is the quick start command:
 
 ```bash
-conda env create -f amadeusGPT.yml
+conda env create -f conda/amadeusGPT.yml
```
Comment (MMathisLab) [inline README.md:39]: drop conda/

File: amadeusgpt/analysis_objects/relationship.py [hunk 3]
```diff
@@ -48,13 +48,14 @@ def calc_angle_between_2d_coordinate_systems(cs1, cs2):
...
-def get_pairwise_distance(arr1, arr2):
+def get_pairwise_distance(arr1: np.ndarray, arr2: np.ndarray):
     # we want to make sure this uses a fast implementation
-    # (n_frame,  n_kpts, 2)
+    # arr: (n_frame,  n_kpts, 2)
     assert len(arr1.shape) == 3 and len(arr2.shape) == 3
     # pariwise distance (n_frames, n_kpts, n_kpts)
     pairwise_distances = np.ones((arr1.shape[0], arr1.shape[1], arr2.shape[1])) * 100000
     for frame_id in range(arr1.shape[0]):
+        # should we use the mean of all keypooints for the distance?
```
Comment (MMathisLab) [inline amadeusgpt/analysis_objects/relationship.py:58]: spelling

**Comment by yeshaokai** [comment]
@MMathisLab incorporated your suggestions

**Review by MMathisLab — APPROVED** [review]

**Pull request merged**